diff --git a/node_modules/isomorphic-git/index.cjs b/node_modules/isomorphic-git/index.cjs
index ae4d526..9d9b046 100644
--- a/node_modules/isomorphic-git/index.cjs
+++ b/node_modules/isomorphic-git/index.cjs
@@ -2935,38 +2935,66 @@ async function _readObject({
     return result
   }
 
-  // BEHOLD! THE ONLY TIME I'VE EVER WANTED TO USE A CASE STATEMENT WITH FOLLOWTHROUGH!
-  // eslint-ignore
-  /* eslint-disable no-fallthrough */
-  switch (result.format) {
-    case 'deflated': {
-      result.object = Buffer.from(await inflate(result.object));
-      result.format = 'wrapped';
-    }
-    case 'wrapped': {
-      if (format === 'wrapped' && result.format === 'wrapped') {
-        return result
-      }
-      const sha = await shasum(result.object);
-      if (sha !== oid) {
-        throw new InternalError(
-          `SHA check failed! Expected ${oid}, computed ${sha}`
-        )
-      }
-      const { object, type } = GitObject.unwrap(result.object);
-      result.type = type;
-      result.object = object;
-      result.format = 'content';
-    }
-    case 'content': {
-      if (format === 'content') return result
-      break
+  // // BEHOLD! THE ONLY TIME I'VE EVER WANTED TO USE A CASE STATEMENT WITH FOLLOWTHROUGH!
+  // // eslint-ignore
+  // /* eslint-disable no-fallthrough */
+  // switch (result.format) {
+  //   case 'deflated': {
+  //     result.object = Buffer.from(await inflate(result.object));
+  //     result.format = 'wrapped';
+  //   }
+  //   case 'wrapped': {
+  //     if (format === 'wrapped' && result.format === 'wrapped') {
+  //       return result
+  //     }
+  //     const sha = await shasum(result.object);
+  //     if (sha !== oid) {
+  //       throw new InternalError(
+  //         `SHA check failed! Expected ${oid}, computed ${sha}`
+  //       )
+  //     }
+  //     const { object, type } = GitObject.unwrap(result.object);
+  //     result.type = type;
+  //     result.object = object;
+  //     result.format = 'content';
+  //   }
+  //   case 'content': {
+  //     if (format === 'content') return result
+  //     break
+  //   }
+  //   default: {
+  //     throw new InternalError(`invalid format "${result.format}"`)
+  //   }
+  // }
+  // /* eslint-enable no-fallthrough */
+
+  /* manually patching https://github.com/isomorphic-git/isomorphic-git/pull/1247 */
+  /* note: this is commonjs export of this package, usually ignored in favour of index.js */
+  if (result.format === 'deflated') {
+    result.object = Buffer.from(await inflate(result.object))
+    result.format = 'wrapped'
+  }
+
+  if (result.format === 'wrapped') {
+    if (format === 'wrapped' && result.format === 'wrapped') {
+      return result
     }
-    default: {
-      throw new InternalError(`invalid format "${result.format}"`)
+    const sha = await shasum(result.object)
+    if (sha !== oid) {
+      throw new InternalError(
+        `SHA check failed! Expected ${oid}, computed ${sha}`
+      )
     }
+    const { object, type } = GitObject.unwrap(result.object)
+    result.type = type
+    result.object = object
+    result.format = 'content'
+  }
+
+  if (result.format === 'content') {
+    if (format === 'content') return result
+    return
   }
-  /* eslint-enable no-fallthrough */
 }
 
 class AlreadyExistsError extends BaseError {
@@ -10337,7 +10365,8 @@ async function _log({ fs, cache, gitdir, ref, depth, since }) {
   const oid = await GitRefManager.resolve({ fs, gitdir, ref });
   const tips = [await _readCommit({ fs, cache, gitdir, oid })];
 
-  while (true) {
+  // https://github.com/isomorphic-git/isomorphic-git/pull/1248
+  while (tips.length > 0) {
     const commit = tips.pop();
 
     // Stop the log if we've hit the age limit
diff --git a/node_modules/isomorphic-git/index.js b/node_modules/isomorphic-git/index.js
index 8335d52..6a6ebd4 100644
--- a/node_modules/isomorphic-git/index.js
+++ b/node_modules/isomorphic-git/index.js
@@ -2929,38 +2929,66 @@ async function _readObject({
     return result
   }
 
-  // BEHOLD! THE ONLY TIME I'VE EVER WANTED TO USE A CASE STATEMENT WITH FOLLOWTHROUGH!
-  // eslint-ignore
-  /* eslint-disable no-fallthrough */
-  switch (result.format) {
-    case 'deflated': {
-      result.object = Buffer.from(await inflate(result.object));
-      result.format = 'wrapped';
-    }
-    case 'wrapped': {
-      if (format === 'wrapped' && result.format === 'wrapped') {
-        return result
-      }
-      const sha = await shasum(result.object);
-      if (sha !== oid) {
-        throw new InternalError(
-          `SHA check failed! Expected ${oid}, computed ${sha}`
-        )
-      }
-      const { object, type } = GitObject.unwrap(result.object);
-      result.type = type;
-      result.object = object;
-      result.format = 'content';
-    }
-    case 'content': {
-      if (format === 'content') return result
-      break
+  // // BEHOLD! THE ONLY TIME I'VE EVER WANTED TO USE A CASE STATEMENT WITH FOLLOWTHROUGH!
+  // // eslint-ignore
+  // /* eslint-disable no-fallthrough */
+  // switch (result.format) {
+  //   case 'deflated': {
+  //     result.object = Buffer.from(await inflate(result.object));
+  //     result.format = 'wrapped';
+  //   }
+  //   case 'wrapped': {
+  //     if (format === 'wrapped' && result.format === 'wrapped') {
+  //       return result
+  //     }
+  //     const sha = await shasum(result.object);
+  //     if (sha !== oid) {
+  //       throw new InternalError(
+  //         `SHA check failed! Expected ${oid}, computed ${sha}`
+  //       )
+  //     }
+  //     const { object, type } = GitObject.unwrap(result.object);
+  //     result.type = type;
+  //     result.object = object;
+  //     result.format = 'content';
+  //   }
+  //   case 'content': {
+  //     if (format === 'content') return result
+  //     break
+  //   }
+  //   default: {
+  //     throw new InternalError(`invalid format "${result.format}"`)
+  //   }
+  // }
+  // /* eslint-enable no-fallthrough */
+
+  /* manually patching https://github.com/isomorphic-git/isomorphic-git/pull/1247 */
+  /* note: this is es module export of this package, usually used instead of index.cjs */
+  if (result.format === 'deflated') {
+    result.object = Buffer.from(await inflate(result.object))
+    result.format = 'wrapped'
+  }
+
+  if (result.format === 'wrapped') {
+    if (format === 'wrapped' && result.format === 'wrapped') {
+      return result
     }
-    default: {
-      throw new InternalError(`invalid format "${result.format}"`)
+    const sha = await shasum(result.object)
+    if (sha !== oid) {
+      throw new InternalError(
+        `SHA check failed! Expected ${oid}, computed ${sha}`
+      )
     }
+    const { object, type } = GitObject.unwrap(result.object)
+    result.type = type
+    result.object = object
+    result.format = 'content'
+  }
+
+  if (result.format === 'content') {
+    if (format === 'content') return result
+    return
   }
-  /* eslint-enable no-fallthrough */
 }
 
 class AlreadyExistsError extends BaseError {
@@ -10331,7 +10359,8 @@ async function _log({ fs, cache, gitdir, ref, depth, since }) {
   const oid = await GitRefManager.resolve({ fs, gitdir, ref });
   const tips = [await _readCommit({ fs, cache, gitdir, oid })];
 
-  while (true) {
+  // https://github.com/isomorphic-git/isomorphic-git/pull/1248
+  while (tips.length > 0) {
     const commit = tips.pop();
 
     // Stop the log if we've hit the age limit
